tests/gocase/unit/introspection/replication_test.go

```diff
@@ -0,0 +1,349 @@
+		require.Equal(t, "b", slave2Client.Get(ctx, "a").Val())
+	})
+}
+
+func TestReplicationContinueRunning(t *testing.T) {
+	master := util.StartServer(t, map[string]string{})
+	defer master.Close()
+	masterClient := master.NewClient()
+	defer func() { require.NoError(t, masterClient.Close()) }()
+
+	slave := util.StartServer(t, map[string]string{})
+	defer slave.Close()
+	slaveClient := slave.NewClient()
+	defer func() { require.NoError(t, slaveClient.Close()) }()
+
+	util.SlaveOf(t, slaveClient, master)
+	util.WaitForSync(t, slaveClient)
+
+	t.Run("Master doesn't pause replicating with replicas, #346", func(t *testing.T) {
+		ctx := context.TODO()
+		// In #346, we find a bug, if one command contains more than special
+		// number updates, master won't send replication stream to replicas.
+		masterClient.HSet(ctx, "myhash", 0, 0, 1, 1, 2, 2, 3, 3, 4, 4, 5, 5, 6, 6, 7, 7, 8, 8, 9, 9,
+			"a", "a", "b", "b", "c", "c", "d", "d", "e", "e", "f", "f", "g", "g", "h", "h", "i", "i", "j", "j", "k", "k")
+		require.EqualValues(t, 21, masterClient.HLen(ctx, "myhash").Val())
+		util.WaitForOffsetSync(t, masterClient, slaveClient)
+		require.Equal(t, "1", slaveClient.HGet(ctx, "myhash", "1").Val())
+		require.Equal(t, "a", slaveClient.HGet(ctx, "myhash", "a").Val())
+	})
+}
+
+func TestReplicationChangePassword(t *testing.T) {
+	master := util.StartServer(t, map[string]string{})
+	defer master.Close()
+	masterClient := master.NewClient()
+	defer func() { require.NoError(t, masterClient.Close()) }()
+
+	slave := util.StartServer(t, map[string]string{})
+	defer slave.Close()
+	slaveClient := slave.NewClient()
+	defer func() { require.NoError(t, slaveClient.Close()) }()
+
+	util.SlaveOf(t, slaveClient, master)
+	util.WaitForSync(t, slaveClient)
+
+	t.Run("Slave can re-sync with master after password change", func(t *testing.T) {
+		ctx := context.TODO()
+		require.Contains(t, slaveClient.Info(ctx, "replication").String(), "role:slave")
+		masterReplicationInfo := masterClient.Info(ctx, "replication").String()
+		require.Contains(t, masterReplicationInfo, "role:master")
+		require.Contains(t, masterReplicationInfo, slave.Host())
+		require.Contains(t, masterReplicationInfo, strconv.Itoa(int(slave.Port())))
+
+		// Change password and break repl connection
+		require.NoError(t, masterClient.ConfigSet(ctx, "requirepass", "pass").Err())
+		require.NoError(t, slaveClient.ConfigSet(ctx, "requirepass", "pass").Err())
+		require.NoError(t, slaveClient.ConfigSet(ctx, "masterauth", "pass").Err())
+
+		killedSlaveCount, err := masterClient.ClientKillByFilter(ctx, "type", "slave").Result()
+		require.NoError(t, err)
+		require.Greater(t, killedSlaveCount, int64(0))
+
+		util.WaitForSync(t, slaveClient)
+		masterReplicationInfo = masterClient.Info(ctx, "replication").String()
+		require.Contains(t, masterReplicationInfo, "role:master")
+		require.Contains(t, masterReplicationInfo, slave.Host())
+		require.Contains(t, masterReplicationInfo, strconv.Itoa(int(slave.Port())))
```

Comment:
This seems not quite equal to:

```tcl
            set ret [$master client list]
            set lines [split $ret "\n"]
            global conn_addr
            foreach  line $lines {
                if {[string match "*cmd=psync*" $line]} {
                    set list [split $line " "]
                    foreach str $list {
                        if {[string match "*addr=*" $str]} {
                            set conn_addr [string range $str 5 end]
                        }
                    }
                }
            }

            $master client kill $conn_addr
            wait_for_sync $slave
            catch {$master info replication} var
            assert_match {*role:master*[$slave_ip]*[$slave_port]*} $var
```

any reason?

Reply:
Kvrocks supported using the type to kill the slave connection, the old implementation is a bit wordy and tricky.

Reply:
Get it.